test/onnx/test_autograd_funs.py

```diff
@@ -0,0 +1,299 @@
+# Owner(s): ["module: onnx"]
+
+from test_pytorch_common import TestCase, run_tests
+
+import torch
+import torch.onnx
+from torch.onnx import (utils,
+                        OperatorExportTypes,
+                        TrainingMode,
+                        register_custom_op_symbolic,
+                        unregister_custom_op_symbolic)
+from torch.onnx.symbolic_helper import (_set_opset_version,
+                                        _set_operator_export_type,
+                                        _set_onnx_shape_inference,
+                                        _unpack_list,
+                                        parse_args)
+import torch.utils.cpp_extension
+from autograd_helper import CustomFunction as CustomFunction2
+from test_pytorch_common import (skipIfUnsupportedMinOpsetVersion,
+                                 skipIfUnsupportedMaxOpsetVersion)
+from verify import verify
+
+import torchvision
+
+import onnx
+
+import io
+import copy
+import unittest
+
+skip = unittest.skip
+
+
+class _BaseTestCase(TestCase):
+
+    def setUp(self):
+        torch.manual_seed(0)
+        if torch.cuda.is_available():
+            torch.cuda.manual_seed_all(0)
+
+    def _model_to_graph(self, model, input,
+                        do_constant_folding=True,
+                        training=TrainingMode.EVAL,
+                        operator_export_type=OperatorExportTypes.ONNX,
+                        input_names=None,
+                        dynamic_axes=None,
+                        inline_autograd=False):
+        if training == torch.onnx.TrainingMode.TRAINING:
+            model.train()
+        elif training == torch.onnx.TrainingMode.EVAL:
+            model.eval()
+        utils._validate_dynamic_axes(dynamic_axes, model, None, None)
+        graph, params_dict, torch_out = utils._model_to_graph(model, input,
+                                                              do_constant_folding=do_constant_folding,
+                                                              _disable_torch_constant_prop=True,
+                                                              operator_export_type=operator_export_type,
+                                                              training=training,
+                                                              input_names=input_names,
+                                                              dynamic_axes=dynamic_axes,
+                                                              inline_autograd=inline_autograd)
+        return graph, params_dict, torch_out
+
+class TestAutogradFuns_opset9(_BaseTestCase):
+    opset_version = 9
```

Comment:
Why is it helpful to test multiple opset versions?
If the logic being tested is independent of opset version, then I suggest we just test with a single opset version (maybe just use the default rather than specifying one).